pkg/virt-controller/watch/replicaset.go

```diff
@@ -351,7 +351,7 @@ func (c *VMIReplicaSet) scale(rs *virtv1.VirtualMachineInstanceReplicaSet, vmis
 // filterActiveVMIs takes a list of VMIs and returns all VMIs which are not in a final state and not terminating
 func (c *VMIReplicaSet) filterActiveVMIs(vmis []*virtv1.VirtualMachineInstance) []*virtv1.VirtualMachineInstance {
 	return filter(vmis, func(vmi *virtv1.VirtualMachineInstance) bool {
-		return !vmi.IsFinal() && vmi.DeletionTimestamp == nil
+		return !vmi.IsFinal() && vmi.DeletionTimestamp == nil && !controller.NewVirtualMachineInstanceConditionManager().HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceConditionType(k8score.PodReady), k8score.ConditionFalse)
```

Comment:
I think what is happening in the k8s case is that the NoExecutionTaintManger is deleting all pods on the unresponsive node. The pods are then still marked as ready (at least if there is no explicit readiness probe specified) but are in the terminating phase. I think that this is the edge case which we are not detecting  right now.

Reply:
Sorry for the late response to the answer. What you said is that the podReady part of vmi is the part that remains ready even if the pod is terminating. I think the podReady part needs to be changed. So I think it needs to be corrected for the part marked as ready. Does your idea match? If not, I think filterActiveVMIs should see the actual pod status and judge it.

Reply:
Which one do you think is the right direction for you?